### PR TITLE
review form added

### PR DESCRIPTION
**Pull request by abrelle**

### CODE REVIEW

**Review by GVabal — APPROVED** [review]

File: src/app/lecturer-page/lecturer-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'app-lecturer-page',
+  templateUrl: './lecturer-page.component.html',
+  styleUrls: ['./lecturer-page.component.scss']
+})
+export class LecturerPageComponent implements OnInit {
+
+  reviewForm = new FormControl('');
```
Comment (pauliusnet) [inline src/app/lecturer-page/lecturer-page.component.ts]:
```suggestion
```
not needed? :?

File: src/app/lecturer-page/lecturer-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (pauliusnet) [inline src/app/lecturer-page/lecturer-page.component.ts]:
```suggestion
```

remove if not used :)

File: src/app/lecturer-page/student-review-form/student-review-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+  public communicationCharsRemaining$: Observable<number> | undefined;
+  public abilityToLearnCharsRemaining$: Observable<number> | undefined;
+  public extraMileCharsRemaining$: Observable<number> | undefined;
+  public motivationCharsRemaining$: Observable<number> | undefined;
+  public directionCharsRemaining$: Observable<number> | undefined;
+  public overallCharsRemaining$: Observable<number> | undefined;
+
+  defaultSliderValue = 5;
+  maxChars = 255;
```
Comment (pauliusnet) [inline src/app/lecturer-page/student-review-form/student-review-form.component.ts]:
```suggestion
  public readonly defaultSliderValue = 5;
  public readonly maxChars = 255;
```
and move these to the very top of the component as these are constants :)

File: src/app/lecturer-page/student-review-form/student-review-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    return this.reviewForm.get('overallComment') as FormControl;
+  }
+
+  get overallGrade(): FormControl {
+    return this.reviewForm.get('overallGrade') as FormControl;
+  }
+
+  submitForm(): void{
+    if (this.reviewForm.valid){
+      this.reviewService.addReview(this.reviewForm.value).subscribe();
```
Comment (pauliusnet) [inline src/app/lecturer-page/student-review-form/student-review-form.component.ts]: no more redux pattern? :)

**Review by pauliusnet — CHANGES_REQUESTED** [review]
Good job!:)

**Review by dziugas306 — APPROVED** [review]

**Review by dziugas306 — APPROVED** [review]

File: src/app/store/reviews/reviews.effects.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { switchMap, map, catchError } from 'rxjs/operators';
+import { ReviewService } from 'src/app/service/review.service';
+import { addReview, addReviewFailure, addReviewSuccess } from './reviews.actions';
+
+
+
+@Injectable()
+export class ReviewsEffects {
+
+  constructor(private actions$: Actions, private reviewService$: ReviewService) {}
```
Comment (GVabal) [inline src/app/store/reviews/reviews.effects.ts]: reviewService ne observable, nenaudočiau $ pavadinimo gale

**Review by GVabal — APPROVED** [review]

**Pull request merged**